File: api/Entities/Admin/CandidateFlightGrp.cs
using System.ComponentModel.DataAnnotations;
using api.Entities.Deployments;

namespace api.Entities.Admin
{
    public class CandidateFlightGrp: BaseEntity
    {
        public DateTime DateOfFlight { get; set; }
        public int OrderNo { get; set; }
        public int CustomerId {get; set;}
        public string AirlineName { get; set; }
        public string FlightNo { get; set; }
        public string AirportOfBoarding { get; set; }
        public string AirportOfDestination { get; set; }
        public DateTime ETD_Boarding { get; set; }
        public DateTime ETA_Destination { get; set; }
        [MaxLength(18)]
        public string ETD_BoardingString { get; set; }
        [MaxLength(18)]
        public string ETA_DestinationString { get; set; }
        
        public string AirportVia { get; set; }
        public string FightNoVia { get; set; }
        public DateTime? ETA_Via { get; set; }
        public DateTime? ETD_Via { get; set; }

        [MaxLength(18)]
        public string ETA_ViaString { get; set; }
        [MaxLength(18)]
        public string ETD_ViaString { get; set; }

        [MaxLength(250)]
        public string FullPath {get; set;}
        public ICollection<CandidateFlightItem> CandidateFlightItems { get; set; }
        
    }
}

File: client/src/app/_models/process/candidateFlightGrp.ts
import { ICandiateFlightItem } from "./candidateFlightItem";


export interface ICandidateFlightGrp
{
    id: number;
    dateOfFlight: Date;
    orderNo: number;
    customerId: number;

    airlineName: string;
    flightNo: string;
    airportOfBoarding: string;
    airportOfDestination: string;
    etD_Boarding: Date;
    etA_Destination: Date;
    etD_BoardingString: string;
    etA_DestinationString: string;
    
    airportVia: string;
    flightNoVia: string;
    etA_Via?: Date | null;
    etD_Via?: Date | null;
    etA_ViaString: string;
    etD_ViaString: string;
    
    fullPath: string;

    fileToUpload: File|null;
    candidateFlightItems: ICandiateFlightItem[];

}

export class CandidateFlightGrp implements ICandidateFlightGrp
{
    id=0;
    dateOfFlight=new Date;
    orderNo=0;
    customerId=0;
    
    airlineName = '';   
    flightNo = ''; 
    airportOfBoarding = '';
    flightNoVia = ''; 
    airportOfDestination = '';
    etD_Boarding= new Date;
    etA_Destination= new Date;
    etA_DestinationString='';
    etD_BoardingString='';
    airportVia = '';
    etD_Via= null;    //new Date;
    etA_Via= null;  // new Date;
    etD_ViaString='';
    etA_ViaString= '';
    
    fullPath='';
    fileToUpload= null;
    candidateFlightItems:ICandiateFlightItem[]=[];
}